### ISSUE TITLE
Add a vignette with examples for the two susceptibility-related matrices

### ISSUE THREAD

**Issue by avallecam** [epiverse-trace/finalsize#128]
After this https://github.com/epiverse-trace/finalsize/issues/119#issuecomment-1384492391 it would be nice to have in a new place a set of two or three examples of these, that would help to complement the [coloured-box definition](https://epiverse-trace.github.io/finalsize/articles/finalsize.html#why-susceptibility-and-p_susceptibility-are-matrices) in Get started vignette.

I drafted some code with five case scenarios describing the population and the two susceptibility-related matrices needed in `final_size()`.

I am not used to matrix objects, so I started with dataframes and transformed them into matrix objects at the end.

This is also a personal exercise to check if I got it correctly, so let me know if there is any mistake.

```
# vignette draft ------------------------------------------------------------

if (!require("pacman")) install.packages("pacman")
#> Loading required package: pacman
pacman::p_load(dplyr)
pacman::p_load(tibble)

# case 1: uniform susceptibility --------------------------------------------------

#' in the population,
#' all individuals from all age groups
#' have 80% susceptibility to the infection

# susceptibility matrix
susceptibility <- 
  tibble(age_group = c("[0,5)","[5,18)","[18,40)","[40,65)","65+"),
         susceptible = c(0.8, 0.8, 0.8, 0.8, 0.8)) %>% 
  column_to_rownames(var = "age_group") %>% 
  as.matrix()

susceptibility
#>         susceptible
#> [0,5)           0.8
#> [5,18)          0.8
#> [18,40)         0.8
#> [40,65)         0.8
#> 65+             0.8

# demography-susceptibility distribution matrix
p_susceptibility <- 
  tibble(age_group = c("[0,5)","[5,18)","[18,40)","[40,65)","65+"),
         susceptible = c(1.0, 1.0, 1.0, 1.0, 1.0)) %>% 
  column_to_rownames(var = "age_group") %>% 
  as.matrix()

p_susceptibility
#>         susceptible
#> [0,5)             1
#> [5,18)            1
#> [18,40)           1
#> [40,65)           1
#> 65+               1

# heterogeneous susceptibility --------------------------------------------

## case 2: only between groups -----------------------------------------------------

#' in the population,
#' there is different
#' susceptibility to the infection
#' between individuals of different age groups
#' from 20% (infants) to 100% (65+)

# susceptibility matrix
susceptibility <- 
  tibble(age_group = c("[0,5)","[5,18)","[18,40)","[40,65)","65+"),
         susceptible = c(0.2, 0.5, 0.6, 0.9, 1.0)) %>% 
  column_to_rownames(var = "age_group") %>% 
  as.matrix()

susceptibility
#>         susceptible
#> [0,5)           0.2
#> [5,18)          0.5
#> [18,40)         0.6
#> [40,65)         0.9
#> 65+             1.0

# demography-susceptibility distribution matrix
p_susceptibility <- 
  tibble(age_group = c("[0,5)","[5,18)","[18,40)","[40,65)","65+"),
         susceptible = c(1.0, 1.0, 1.0, 1.0, 1.0)) %>% 
  column_to_rownames(var = "age_group") %>% 
  as.matrix()

p_susceptibility
#>         susceptible
#> [0,5)             1
#> [5,18)            1
#> [18,40)           1
#> [40,65)           1
#> 65+               1

## case 3: within and between groups --------------------------------------------

#' in the population,
#' there is different
#' susceptibility to the infection
#' between individuals of different age groups
#' from 20% (infants) to 100% (65+)
#' and
#' within individuals of the same age group
#' due the immunization effect of 25%
#' to the 40% of each of the age groups

immunization_effect <- 0.25

# susceptibility matrix
susceptibility <- 
  tibble(age_group = c("[0,5)","[5,18)","[18,40)","[40,65)","65+"),
         unimmunised = c(0.2, 0.5, 0.6, 0.9, 1.0)) %>% 
  mutate(immunised = unimmunised * (1 - immunization_effect)) %>%
  column_to_rownames(var = "age_group") %>% 
  as.matrix()

susceptibility
#>         unimmunised immunised
#> [0,5)           0.2     0.150
#> [5,18)          0.5     0.375
#> [18,40)         0.6     0.450
#> [40,65)         0.9     0.675
#> 65+             1.0     0.750

# demography-susceptibility distribution matrix
p_susceptibility <- 
  tibble(age_group = c("[0,5)","[5,18)","[18,40)","[40,65)","65+"),
         unimmunised = c(0.6, 0.6, 0.6, 0.6, 0.6)) %>% 
  mutate(immunised = 1 - unimmunised) %>% 
  column_to_rownames(var = "age_group") %>% 
  as.matrix()

p_susceptibility
#>         unimmunised immunised
#> [0,5)           0.6       0.4
#> [5,18)          0.6       0.4
#> [18,40)         0.6       0.4
#> [40,65)         0.6       0.4
#> 65+             0.6       0.4


## case 4: within in different proportion ------------------------------------------

#' in the population,
#' there is different
#' susceptibility to the infection
#' between individuals of different age groups
#' from 20% (infants) to 100% (65+)
#' and
#' within individuals of the same age group
#' due the immunization effect of 25%
#' to different proportions for each of the age groups
#' immunisation increases with age
#' from 20% (infants) to 90% (65+)

immunization_effect <- 0.25

# susceptibility matrix
susceptibility <- 
  tibble(age_group = c("[0,5)","[5,18)","[18,40)","[40,65)","65+"),
         unimmunised = c(0.2, 0.5, 0.6, 0.9, 1.0)) %>% 
  mutate(immunised = unimmunised * (1 - immunization_effect)) %>%
  column_to_rownames(var = "age_group") %>% 
  as.matrix()

susceptibility
#>         unimmunised immunised
#> [0,5)           0.2     0.150
#> [5,18)          0.5     0.375
#> [18,40)         0.6     0.450
#> [40,65)         0.9     0.675
#> 65+             1.0     0.750

# demography-susceptibility distribution matrix
p_susceptibility <- 
  tibble(age_group = c("[0,5)","[5,18)","[18,40)","[40,65)","65+"),
         unimmunised = c(0.8, 0.6, 0.4, 0.3, 0.1)) %>% 
  mutate(immunised = 1 - unimmunised) %>% 
  column_to_rownames(var = "age_group") %>% 
  as.matrix()

p_susceptibility
#>         unimmunised immunised
#> [0,5)           0.8       0.2
#> [5,18)          0.6       0.4
#> [18,40)         0.4       0.6
#> [40,65)         0.3       0.7
#> 65+             0.1       0.9

## case 5: within three groups -----------------------------------------------------

#' in the population,
#' there is different
#' susceptibility to the infection
#' between individuals of different age groups
#' from 20% (infants) to 100% (65+)
#' and
#' within individuals of the same age group
#' due the immunization effect of 25%
#' to the 40% of each of the age groups
#' and
#' due to 10% of individuals
#' in each of the age groups
#' non immunised and non exposed
#' to similar infections
#' with 100% susceptibility

immunization_effect <- 0.25

# susceptibility matrix
susceptibility <- 
  tibble(age_group = c("[0,5)","[5,18)","[18,40)","[40,65)","65+"),
         susceptible = c(1.0, 1.0, 1.0, 1.0, 1.0),
         unimmunised = c(0.2, 0.5, 0.6, 0.9, 1.0)) %>% 
  mutate(immunised = unimmunised * (1 - immunization_effect)) %>%
  column_to_rownames(var = "age_group") %>% 
  as.matrix()

susceptibility
#>         susceptible unimmunised immunised
#> [0,5)             1         0.2     0.150
#> [5,18)            1         0.5     0.375
#> [18,40)           1         0.6     0.450
#> [40,65)           1         0.9     0.675
#> 65+               1         1.0     0.750

# demography-susceptibility distribution matrix
p_susceptibility <- 
  tibble(age_group = c("[0,5)","[5,18)","[18,40)","[40,65)","65+"),
         susceptible = c(0.1, 0.1, 0.1, 0.1, 0.1),
         unimmunised = c(0.6, 0.6, 0.6, 0.6, 0.6)) %>% 
  mutate(immunised = 1 - unimmunised - susceptible) %>% 
  column_to_rownames(var = "age_group") %>% 
  as.matrix()

p_susceptibility
#>         susceptible unimmunised immunised
#> [0,5)           0.1         0.6       0.3
#> [5,18)          0.1         0.6       0.3
#> [18,40)         0.1         0.6       0.3
#> [40,65)         0.1         0.6       0.3
#> 65+             0.1         0.6       0.3

```

**Comment by pratikunterwegs** [comment]
Would it be easier for you implement this on a branch off of `dev/fixes` and add this as a PR back into `dev/fixes`?

**Comment by avallecam** [comment]
Sure!

**Comment by pratikunterwegs** [comment]
Just thinking about this for a bit, and writing down what I mentioned on Slack: I would suggest that you make a feature branch off your fork's `main`, implement the vignette on that feature branch, and make a PR from that feature branch to the `epiverse-trace/finalsize` (original = upstream) `main`. That means there will be two PRs open to `main`, #121 and the new one. Once #121 is merged, you can update your fork's feature branch by rebasing on the upstream `main`. Once the fork feature branch is updated, you can squash merge it into the upstream `main`. Then update your fork `main` with the upstream `main`.

The downside I see with this is that this issue relates strongly to #121 and should be linked in some way, and this particular forking workflow won't really allow for that.
